### PR TITLE
[system-probe] Use `bpf_get_prandom_u32` for cookie generation on CO-RE

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

Uses the bpf helper `bpf_get_prandom_u32` instead of the home baked random num generation on CO-RE.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/tracer/stats.h [hunk 1]
```diff
@@ -21,6 +21,11 @@ static __always_inline __u64 offset_rtt_var();
 #endif
 
 static __always_inline conn_stats_ts_t *get_conn_stats(conn_tuple_t *t, struct sock *sk) {
+    conn_stats_ts_t *cst = bpf_map_lookup_elem(&conn_stats, t);
+    if (cst) {
```
Comment (brycekahle) [inline pkg/network/ebpf/c/tracer/stats.h:25]: Is this to try and save the call to `get_sk_cookie`?

Comment (hmahmood) [reply pkg/network/ebpf/c/tracer/stats.h:25]: Yes, and a call to `bpf_map_update_elem` since that is also causing a lot of errors in bpf telemetry.

Comment (brycekahle) [reply pkg/network/ebpf/c/tracer/stats.h:25]: `BPF_EXIST` shouldn't really be treated like an error, we should remedy that.

Comment (hmahmood) [reply pkg/network/ebpf/c/tracer/stats.h:25]: @usamasaqib

**Pull request closed without merging**